oauth2_http/java/com/google/auth/oauth2/S2A.java

```diff
@@ -143,6 +143,11 @@ private S2AConfig getS2AConfigFromMDS() {
       try {
         HttpResponse response = request.execute();
         if (!response.isSuccessStatusCode()) {
+          int statusCode = response.getStatusCode();
```

Comment:
fyi: @lqiu96 @zhumin8 , after chatting with @xmenxk (authored the Go implementation: https://github.com/googleapis/google-cloud-go/blob/main/auth/internal/transport/s2a.go), he mentioned that we shouldn't retry if the endpoint doesn't exist. This logic exists in Go impl, so adding it here as well.

Reply:
I lean toward switching to HttpRequest's built-in retry handlers, logic like to customize the retry behavior based on return code can go into `HttpUnsuccessfulResponseHandler`. See other comment for details.

Reply:
> we shouldn't retry if the endpoint doesn't exist

This means that we don't retry on 404 status code, right?

Reply:
I am leaning toward switching to HttpRequest's built-in retry handlers, logic like to customize the retry behavior based on return code can go into HttpUnsuccessfulResponseHandler. See other comment for details.
> SG! I'll address your other comment.

This means that we don't retry only on 404 status code, right?
> Yes, we only want to retry on 404.